log.h

```diff
@@ -1,15 +1,27 @@
-#pragma once
-
-#define LOG_ERROR(fmt, ...) \
-  { printf("ERROR - "), printf(fmt, ##__VA_ARGS__); }
-#define LOG_WARN(fmt, ...) \
-  { printf("WARNING - "), printf(fmt, ##__VA_ARGS__); }
-#ifdef _DEBUG
-#  define LOG_INFO(fmt, ...) \
-    { printf("INFO - "), printf(fmt, ##__VA_ARGS__); }
-#  define LOG_DEBUG(fmt, ...) \
-    { printf("DEBUG - "), printf(fmt, ##__VA_ARGS__); }
-#else
-#  define LOG_INFO(fmt, ...)
-#  define LOG_DEBUG(fmt, ...)
-#endif
+#pragma once
+
+#include <stdarg.h>
+#include <stdio.h>
+
+void proxy_log_set_error(void (*func)(const char *fmt, va_list args));
+void proxy_log_set_warn(void (*func)(const char *fmt, va_list args));
+void proxy_log_set_info(void (*func)(const char *fmt, va_list args));
+void proxy_log_set_debug(void (*func)(const char *fmt, va_list args));

```

Comment:
nit: You might want to include "callback" or "func", so people know you're overriding the callback function.

Reply:
Done.